File: practicas/QytetetRuby/pkg/QytetetRuby-0.0.1/lib/sorpresa.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  class Sorpresa
    def initialize(unTexto, unValor, unTipo)
      @texto = unTexto #testo de carta Sorpresa
      @valor = unValor #valor de sorpresa
      @tipo = unTipo #tipo de sorpresa de TipoSorpresa
    end

    attr_reader:texto
    attr_reader:valor
    attr_reader:tipo

  #forma explicita  
=begin
    def texto()
      return texto
    end

    def valor()
      return valor
    end

    def tipo()
      return tipo
    end
=end

    def to_s()
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end

  end
end

File: practicas/QytetetJava/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author venrra
 */
public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    public Sorpresa(String texto, int valor, TipoSorpresa tipo){
        this.texto = texto;
        this.tipo = tipo;
        this.valor = valor;
    }
    
    public String getTexto(){
        return this.texto;
    }
    
    public TipoSorpresa getTipo(){
        return this.tipo;
    }
    
    public int getValor(){
        return this.valor;
    }
    
    @Override
    public String toString(){
        return "Sorpresa{" + "texto=" + texto + ", valor= " + Integer.toString(valor) + ", tipo=" + tipo +"}";
    }
}
